addon/addon/models/sprite-tree.ts

```diff
@@ -174,7 +188,7 @@ export default class SpriteTree {
       let node = this.lookupNodeByElement(context.element);
       let ancestor = node && node.parent;
       while (ancestor) {
-        if (ancestor.nodeType === SpriteTreeNodeType.Context) {
+        if (ancestor.nodeType.has(SpriteTreeNodeType.Context)) {
           if (result.indexOf(ancestor.model as ContextModel) === -1) {
```

Comment:
We don't have a guarantee that the model is the AnimationContext and not the SpriteModifier if the node has both, I think

Reply:
Right! I missed that there was a difference there even though right now it is only the interface name that differs (i.e. it's just for typing purposes). Internally inside this class it doesn't really seem to matter (we already cast the type anyway and use the nodeType to do the actual filtering) so the question is do we care about the difference here? It seems to mostly function as a way to type the arguments to `addAnimationContext` and such, which still works fine 🤔

Reply:
Externally we use this to check which contexts will transition: 

https://github.com/cardstack/animations-experiment/blob/e3d49589e8b9859d9167ecfd42607b94ea1c74cd/addon/addon/services/animations.ts#L197 

and also get some sprite modifiers here for intermediate sprites:

https://github.com/cardstack/animations-experiment/blob/e3d49589e8b9859d9167ecfd42607b94ea1c74cd/addon/addon/services/animations.ts#L152

Reply:
Yes, but getContextRunlists filters on nodeType and casts everything to a ContextModel is what I meant.

Reply:
On the nested-contexts demo route, clicking "Show level 3" will fail at `context.shouldAnimate` (https://github.com/cardstack/animations-experiment/blob/bad6ad25c1d9bdfd986c73d5b8fd47cf1759d0fb/addon/addon/models/transition-runner.ts#L111) because the passed model is a sprite modifier. I think we do need to have 2 models on a node.

Reply:
I'm not sure we're really casting things, we're just typing them with something like a marker type. The expected structure of a ContextModel and a SpriteModel are both `{ element: Element }`;

Reply:
Ah there's a couple public methods on the SpriteTree that have the model as output which is likely causing this. Taking a look, but it may not be straightforward to solve this (other than possibly "duplicating" element models, i.e. return both models if something is a Sprite as well as a Context).